frontend/app/src/composables/blockchain/balances/index.ts

```diff
@@ -69,10 +65,32 @@ export const useBlockchainBalances = () => {
     }
   };
 
+  const fetch = async (
+    blockchain: Blockchain,
+    ignoreCache = false,
+    periodic = false
+  ): Promise<void> => {
+    const { isLoading } = useStatusStore();
+    const loading = isLoading(chainSection[blockchain]);
+
+    const call = () => handleFetch(blockchain, ignoreCache);
+
+    if (get(loading)) {
+      if (periodic) {
+        return;
+      }
+
+      await until(loading).toBe(false);
```

Comment:
@kelsos, by assuming that all manual refresh all disabled, when there is balance refresh running, this part will queued once for manual refresh (by click the refresh button, or triggered by detect tokens)

Reply:
Can you maybe also add a test for this? just to be sure?

Reply:
absolutely